File: 217_contains_duplicate/C++/217.cpp
#include <iostream>
#include <vector>
#include <unordered_set>

using namespace std;

class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        unordered_set<int> set;

        for (int x : nums)
            if (set.count(x))
                return true;
            else
                set.insert(x);

        return false;
    }
};

void printAnswer(bool answer)
{   
    cout << (answer ? "true" : "false") << '\n';
}

int main()
{
    //vector<int> nums {1, 2, 3, 1};
    //vector<int> nums {1, 2, 3, 4};
    vector<int> nums {1, 1, 1, 3, 3, 4, 3, 2, 4, 2};
    
    Solution solution;
    printAnswer(solution.containsDuplicate(nums));

    return 0;
}

File: 217_contains_duplicate/Python/217.py
from typing import List

class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        mySet = set()

        for num in nums:
            if num in mySet:
                return True
            mySet.add(num)
        
        return False

nums = [1,2,3,1]
print(Solution.containsDuplicate(None, nums))
